ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.remoting.ssl;
+
+import com.google.common.base.Throwables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * Utility functions for creating {@link SSLSocketFactory}s
+ * that are configured with Java trust stores and key stores.
+ */
+public final class SslSocketFactories {
+    private SslSocketFactories() {}
+
+    /**
+     * Create an {@link SSLSocketFactory} from the provided configuration.
```

Comment:
Fine with making these changes, but note that I'm matching the comment style that already exists in the project -- see `com.palantir.trust.TrustStores`. If I make the changes you suggest here, style will be inconsistent with that file.